daemon/kmd/client/wrappers.go

```diff
@@ -59,6 +59,17 @@ func (kcl KMDClient) CreateWallet(walletName []byte, walletDriverName string, wa
 	return
 }
 
+// RenameWallet wraps kmdapi.APIV1POSTWalletRenameRequest
+func (kcl KMDClient) RenameWallet(walletID []byte, newWalletName []byte, walletPassword []byte) (resp kmdapi.APIV1POSTWalletRenameResponse, err error) {
+	req := kmdapi.APIV1POSTWalletRenameRequest{
+		WalletID:       string(walletID),
+		NewWalletName:  string(newWalletName),
+		WalletPassword: string(walletPassword),
```

Comment:
My gut feeling is that this action should require a wallet handle from `/v1/wallet/init` and then passed into this function. Similar to how an export would work. But since you're requesting the wallet password here instead it sort of shortcuts the flow.

I tested it and it all seems to work, but when a wallet doesn't have a password it feels weird that a single call to the KMD API `/v1/wallets/rename` can change the name of the wallet without getting a handle first. Without spending more time looking at other calls though, I am not strongly against this approach, just not sure if it'd be accepted upstream.

Reply:
KMD already supports renames through the API, it is just that `goal wallet` didn't

The types e.g. APIV1POSTWalletRenameRequest were there already, so I opted not to change those and use them as they are.